src/main/scala/org/monarchinitiative/dosdp/DOSDP.scala

```diff
@@ -81,42 +83,61 @@ object DOSDP {
 
 trait PrintfText {
 
-  def text: String
+  def text: Option[String]
 
   def vars: Option[List[String]]
 
   def annotations: Option[List[Annotations]]
 
-  def replaced(bindings: Option[Map[String, SingleValue]]): Option[String] = PrintfText.replaced(this.text, this.vars, bindings, shouldQuote)
+  def replaced(bindings: Option[Map[String, SingleValue]]): Option[String] = PrintfText.replaced(this.text, this.vars, None, bindings, shouldQuote)
 
   def shouldQuote: Boolean
 
 }
 
 object PrintfText {
 
-  def replaced(text: String, vars: Option[List[String]], bindings: Option[Map[String, SingleValue]], quote: Boolean): Option[String] = {
+  def replaced(text: Option[String], vars: Option[List[String]], multi_clause: Option[MultiClausePrintf], bindings: Option[Map[String, SingleValue]], quote: Boolean): Option[String] = {
+    if(text.isDefined) {
+      return replace_text(text, vars, bindings, quote)
+    } else if (multi_clause.isDefined) {
+      val replaced_texts = for {
+        multi_clauses <- multi_clause.toSeq
+        printf_clauses <- multi_clauses.clauses.toSeq
+        printf_clause <- printf_clauses
+        printf_clause_text <- replaced(Some(printf_clause.text), printf_clause.vars, None, bindings, quote)
+        sub_clauses = printf_clause.sub_clauses.getOrElse(List.empty[MultiClausePrintf])
+        sub_clauses_texts = sub_clauses.flatMap(mc => replaced(None, None, Some(mc), bindings, quote))
+        clause_text = (printf_clause_text :: sub_clauses_texts).mkString(multi_clauses.sep.getOrElse(" "))
+      } yield clause_text
+      val result = replaced_texts.filter(_.nonEmpty).mkString(multi_clause.get.sep.getOrElse(" "))
+      val trimmed = result.trim
+      if (trimmed.isEmpty) return None else return Some(trimmed)
+    }
+    None
+  }
+
+  private def replace_text(text: Option[String], vars: Option[List[String]], bindings: Option[Map[String, SingleValue]], quote: Boolean) = {
```

Comment:
It seems like `text` should be `String`, not `Option[String]` (handle missing text outside of this method). Also, I like to add type signatures to method definitions.

Reply:
Based on the [new schema](https://github.com/INCATools/dead_simple_owl_design_patterns/blob/e5d69a6f3e17ec464e6942ccf0f05991694cd671/spec/DOSDP_schema_full.yaml#L135) text or multi_clause can be used to define printf formats. 
```yaml
    oneOf:
      - required: [ text, vars ]
      - required: [ multi_clause ]
```


Text is not a mandatory field now, we can use multi_clause instead of text.